src/analysis.cpp

```diff
@@ -1322,6 +1326,192 @@ PolymerShape::PolymerShape(const json& j, Space& spc) : Analysisbase(spc, "Polym
+    analysis.sasa->init(analysis.spc);
+    std::for_each(first, last, [&analysis](const auto& species) {
+        double area = analysis.sasa->calcSASAOf(analysis.spc, species);
+        analysis.takeSample(area);
+    });
+}
+
+/** @brief samples a sum of SASAs of objects (either a whole group or a particle)
+ *         between first and last
+ * @param first iterator at first particle or group
+ * @param last iterator at last particle or group
+ * @param analysis SASAanalysis object to insert samples into
+ *
+ * @tparam TBegin
+ * @tparam TEnd
+ * */
+template <typename TBegin, typename TEnd>
+void SamplingPolicyBase::sampleTotalSASA(TBegin first, TEnd last, SASAAnalysis& analysis) {
+    analysis.sasa->init(analysis.spc);
+    double area = analysis.sasa->calcSASA(analysis.spc, first, last);
+    analysis.takeSample(area);
+    if (analysis.output_stream) {
+        *analysis.output_stream << fmt::format("{} {:.3f} \n", analysis.getNumberOfSteps(), area);
+    }
+}
+
+void AtomicPolicy::sample(Space& spc, SASAAnalysis& analysis) {
+    auto atoms = spc.findAtoms(atom_id);
+    sampleIndividualSASA(atoms.begin(), atoms.end(), analysis);
+}
+void AtomicPolicy::to_json(json& json_output) const {
+    json_output["atom_name"].push_back(atom_name);
+}
+void AtomicPolicy::from_json(const json& json_input) {
+    atom_name = json_input.at("atom").get<std::string>();
+    atom_id = findAtomByName(atom_name).id();
+}
+
+void MolecularPolicy::sample(Space& spc, SASAAnalysis& analysis) {
+    auto molecules = spc.findMolecules(molecule_id);
+    sampleIndividualSASA(molecules.begin(), molecules.end(), analysis);
+}
+void MolecularPolicy::to_json(json& json_output) const {
+    json_output["molecule"].push_back(molecule_name);
+}
+void MolecularPolicy::from_json(const json& json_input) {
+    molecule_name = json_input.at("molecule").get<std::string>();
+    molecule_id = findMoleculeByName(molecule_name).id();
+}
+
+void AtomsInMoleculePolicy::sample(Space& spc, SASAAnalysis& analysis) {
+
+    auto molecules = spc.findMolecules(molecule_id);
+    for( const auto& molecule : molecules ){
+
+        auto selected_atoms = molecule | ranges::cpp20::views::filter([this, &molecule](const Particle& particle) {
+                                  return selected_indices.count(molecule.getParticleIndex(particle)) > 0; } ) ;
+        sampleTotalSASA(selected_atoms.begin(), selected_atoms.end(), analysis);
+    }
+}
+
+void AtomsInMoleculePolicy::to_json(json& json_output) const {
+    json_output["molecule"].push_back(molecule_name);
+    json_output["atomlist"].push_back(atom_names);
+}
+void AtomsInMoleculePolicy::from_json(const json& json_input) {
+
+    const auto atomlist = json_input.at("atomlist");
+    if(!atomlist.empty()) {
+        for (const auto& atom_specifier : atomlist) {
+            if(atom_specifier.is_string()) {
+                const auto& atom_name = atom_specifier.get<std::string>();
+                atom_names.insert(atom_name);
```

Comment:
This could be a one-liner, `atom_names.insert(atom_specifier);` as we have already asserted that it's a string.